### PR TITLE
test: Add Dev Environment CR to appstudio tier template

### PR DESCRIPTION
**Pull request by xcoulon**
e2e tests for https://github.com/codeready-toolchain/host-operator/pull/735

Signed-off-by: Xavier Coulon <xcoulon@redhat.com>

### CODE REVIEW

File: testsupport/wait/member.go [hunk 1]
```diff
@@ -1965,3 +1966,19 @@ func (a *MemberAwaitility) UpdateConfigMap(t *testing.T, namespace, cmName strin
...
 }
+
+func (a *MemberAwaitility) WaitForEnvironment(t *testing.T, namespace, name string) (*appstudiov1.Environment, error) {
+	var env *appstudiov1.Environment
+	err := wait.Poll(a.RetryInterval, a.Timeout, func() (done bool, err error) {
+		obj := &appstudiov1.Environment{}
+		if err := a.Client.Get(context.TODO(), types.NamespacedName{
+			Namespace: namespace,
+			Name:      name},
+			obj); err != nil {
```
Comment (mfrancisc) [inline testsupport/wait/member.go:1977]: shouldn't it return `false` on error?

```
err != nil {
   if errors.IsNotFound(err) {
      return false, nil
   }
   return false, err
}
```

File: test/e2e/nstemplatetier_test.go [hunk 2]
```diff
@@ -39,6 +39,7 @@ func TestNSTemplateTiers(t *testing.T) {
 		EnsureMUR().
 		RequireConditions(ConditionSet(Default(), ApprovedByAdmin())...).
 		Execute(t).
+		DisableCleanup().
```
Comment (MatousJobanek) [inline test/e2e/nstemplatetier_test.go:42]: :ghost: 
```suggestion
```

File: testsupport/appstudio/api/v1alpha1/README.adoc [hunk 3]
```diff
@@ -0,0 +1,8 @@
+= HOW-TO
+
+Copy content from:
+- https://github.com/redhat-appstudio/application-api/blob/main/api/v1alpha1/environment_types.go
+- https://github.com/redhat-appstudio/application-api/blob/main/api/v1alpha1/snapshotenvironmentbinding_types.go
```
Comment (MatousJobanek) [inline testsupport/appstudio/api/v1alpha1/README.adoc:5]: we don't use `snapshotenvironmentbinding_types.go`, right?
```suggestion
- https://github.com/redhat-appstudio/application-api/blob/main/api/v1alpha1/application_types.go
```

File: testsupport/appstudio/api/v1alpha1/README.adoc [hunk 4]
```diff
@@ -0,0 +1,8 @@
...
+- https://github.com/redhat-appstudio/application-api/blob/main/api/v1alpha1/zz_generated.deepcopy.go
+
+and remove everything that is not related to the `Environment` type.
```
Comment (MatousJobanek) [inline testsupport/appstudio/api/v1alpha1/README.adoc:8]:
```suggestion
and remove everything that is not related to the types: `Environment`, `Application`.
```

File: testsupport/tiers/checks.go [hunk 5]
```diff
@@ -1153,6 +1154,14 @@ func gitOpsServiceLabel() namespaceObjectsCheck {
 	}
 }
 
+func environment(name string) namespaceObjectsCheck {
+	return func(t *testing.T, ns *corev1.Namespace, memberAwait *wait.MemberAwaitility, _ string) {
+		// TODO fix for migration/existing namespaces cases
```
Comment (MatousJobanek) [inline testsupport/tiers/checks.go:1159]: >// TODO fix for migration/existing namespaces cases

what do you mean by that?

**Review by MatousJobanek — COMMENTED** [review]
Looks good - just a few minor comments :+1:

File: test/e2e/nstemplatetier_test.go (hunk 2 above)
Comment (xcoulon) [reply test/e2e/nstemplatetier_test.go:42]: ah right! 🤦‍♂️

File: testsupport/appstudio/api/v1alpha1/README.adoc (hunk 3 above)
Comment (xcoulon) [reply testsupport/appstudio/api/v1alpha1/README.adoc:5]: https://github.com/redhat-appstudio/application-api/blob/main/api/v1alpha1/snapshotenvironmentbinding_types.go contains the definition for [EnvVarPair](https://github.com/redhat-appstudio/application-api/blob/main/api/v1alpha1/snapshotenvironmentbinding_types.go#L96-L103) which is used in [EnvironmentConfiguration](https://github.com/redhat-appstudio/application-api/blob/main/api/v1alpha1/environment_types.go#L112-L115)

File: testsupport/tiers/checks.go (hunk 5 above)
Comment (xcoulon) [reply testsupport/tiers/checks.go:1159]: never mind, it's a remainder of a quick copy/paste of the [gitOpsServiceLabel func](https://github.com/codeready-toolchain/toolchain-e2e/blob/master/testsupport/tiers/checks.go#L1146)

Comment (xcoulon) [reply testsupport/tiers/checks.go]: removed in 473dad2

File: testsupport/wait/member.go (hunk 1 above)
Comment (xcoulon) [reply testsupport/wait/member.go:1977]: yes, indeed! 🤦‍♂️

Comment (xcoulon) [reply testsupport/wait/member.go]: fixed in d52d300

**Review by alexeykazakov — APPROVED** [review]

**Review by mfrancisc — APPROVED** [review]

File: testsupport/appstudio/api/v1alpha1/README.adoc (hunk 3 above)
Comment (MatousJobanek) [reply testsupport/appstudio/api/v1alpha1/README.adoc:5]: ah ok, so you put everything into one file? I missed that :man_facepalming:

**Review by MatousJobanek — APPROVED** [review]

**Comment by xcoulon** [comment]
thanks for merging the PR, @alexeykazakov!

**Pull request merged**